### PR TITLE
samples: drivers: watchdog running on the stm32H7 platforms

### PR DESCRIPTION
**Pull request by FRASTM**
Change the condition to have the samples/drivers/watchdog PASSED on the stm32h7 target boards.

### CODE REVIEW

File: samples/drivers/watchdog/boards/stm32h7_wwdg.overlay [hunk 1]
```diff
@@ -11,10 +11,6 @@
 	};
 };
 
-&rcc {
-	d2ppre1 = <16>;
-};
-
```
Comment (martinjaeger) [inline samples/drivers/watchdog/boards/stm32h7_wwdg.overlay:17]: Can you please state in the commit message why the RCC setting was wrong? You added it a while ago in commit 838cd28195631e039b569f39cdc10cf1c839807f saying it is required.

**Review by martinjaeger — CHANGES_REQUESTED** [review]

File: samples/drivers/watchdog/boards/stm32h7_wwdg.overlay (hunk 1 above)
Comment (FRASTM) [reply samples/drivers/watchdog/boards/stm32h7_wwdg.overlay]: You're right
 even if at that time, the sysclock was much lower (typ. 96MHz) for the stm32h7xx, (now it is 480) there is no need to change it.

File: samples/drivers/watchdog/sample.yaml [hunk 2]
```diff
@@ -48,7 +48,19 @@ tests:
...
+      - stm32h7s78_dk
+      - nucleo_h743zi
+      - stm32h735g_disco
+      - nucleo_h753zi
+      - stm32h750b_dk
+      - stm32h7b3i_dk
+      - stm32h745i_disco
+      - nucleo_h755zi_q
+      - stm32h747i_disco
+      - nucleo_h7a3zi_q
```
Comment (erwango) [inline samples/drivers/watchdog/sample.yaml:63]: What about filtering on `SOC_SERIES_STM32H7X` ?

Comment (FRASTM) [reply samples/drivers/watchdog/sample.yaml]: `filter: dt_compat_enabled("st,stm32-window-watchdog") and CONFIG_SOC_SERIES_STM32H7X`
does not filter correctly
neither
`filter: CONFIG_SOC_SERIES_STM32H7X and CONFIG_WWDG_STM32`

**Comment by FRASTM** [comment]
keep rcc D2PPR1 prescaler  unchanged for the stm32H7 serie.
In the stm32 WWDg driver, enable the watchdog IRQ at the end of the initialization process, once the peripheral is clocked
else it might cause lock (seen on nucleo_h743zi) :
```
*** Booting Zephyr OS build (tainted) v3.7.0-5097-gdfbcea777a10 ***                                                               
Watchdog sample application                                                                                                       
Attempting to test pre-reset callback                                                                                             
[00:00:00.011,000] <dbg> wdt_wwdg_stm32: wwdg_stm32_install_timeout: prescaler: 64                                                
[00:00:00.019,000] <dbg> wdt_wwdg_stm32: wwdg_stm32_install_timeout: Desired WDT: 100000 us                                       
[00:00:00.028,000] <dbg> wdt_wwdg_stm32: wwdg_stm32_install_timeout: Set WDT:     100488 us                                       
Feeding watchdog 5 times                                                                                                          
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Feeding watchdog...                                                                                                               
Waiting for reset...      
```

**Review by martinjaeger — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by mathieuchopstm — DISMISSED** [review]

File: samples/drivers/watchdog/sample.yaml [hunk 3]
```diff
@@ -48,7 +48,18 @@ tests:
   sample.drivers.watchdog.stm32h7_wwdg:
     extra_args: DTC_OVERLAY_FILE=boards/stm32h7_wwdg.overlay
     filter: dt_compat_enabled("st,stm32-window-watchdog")
-    platform_allow: nucleo_h743zi
+    platform_allow:
```
Comment (nashif) [inline samples/drivers/watchdog/sample.yaml:51]: isnt this already covered with the filter above? why the platform allow thing?

Comment (nashif) [reply samples/drivers/watchdog/sample.yaml:51]: also, please add integration_platforms with one platform for PR verification.

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by nashif — APPROVED** [review]

File: samples/drivers/watchdog/sample.yaml (hunk 3 above)
Comment (erwango) [reply samples/drivers/watchdog/sample.yaml:53]: One typical example where https://github.com/zephyrproject-rtos/zephyr/issues/81458 would help.

**Review by erwango — APPROVED** [review]

**Pull request merged**